### PR TITLE
Refresh snapshots docs page

### PR DESCRIPTION
**Pull request by jamiebrynes7**
#### Description
Refreshed the snapshots docs page:

- Removed reference to Playground.
- Tidied up the prose and link to our own glossary.
- Added a code sample.
- Pointed users to tooling for launching deployments before pointing them to the CLI.

### CODE REVIEW

File: docs/reference/concepts/snapshots.md [hunk 1]
```diff
@@ -2,32 +2,52 @@
...
+    template.AddComponent(new Persistence.Snapshot(), WorkerUtils.UnityGameLogic);
+    template.SetReadAccess(WorkerUtils.UnityGameLogic, WorkerUtils.UnityClient, WorkerUtils.MobileClient);
+    template.SetComponentWriteAccess(EntityAcl.ComponentId, WorkerUtils.UnityGameLogic);
 
-### Generate the snapshot
+    // ..and add it to the snapshot.
+    snapshot.AddEntity(template);
 
-To generate the snapshot, in your Unity Editor menu: **SpatialOS** > **Generate snapshot** to open the snapshot generator window, then select `Generate snapshot` there.
+    // WriteToFile operates realtive to your working directory.
```
Comment (paulbalaji) [inline docs/reference/concepts/snapshots.md]:
```suggestion
    // WriteToFile operates relative to your working directory.
```

File: docs/reference/concepts/snapshots.md [hunk 2]
```diff
@@ -2,32 +2,52 @@
...
-
-You use a snapshot as the starting point for your [world](https://docs.improbable.io/reference/<%(Var key="worker_sdk_version")%>/shared/glossary#spatialos-world) when you [deploy](https://docs.improbable.io/reference/<%(Var key="worker_sdk_version")%>/shared/glossary#deploying), [locally](https://docs.improbable.io/reference/<%(Var key="worker_sdk_version")%>/shared/glossary#local-deployment) or [to the cloud](https://docs.improbable.io/reference/<%(Var key="worker_sdk_version")%>/shared/glossary#cloud-deployment).
+You use a snapshot as the starting point for your [world]({{urlRoot}}/reference/glossary#spatialos-world) when you [deploy]({{urlRoot}}/reference/glossary#deploying).
```
Comment (paulbalaji) [inline docs/reference/concepts/snapshots.md]:
```suggestion
You can use a snapshot as the starting point for your [world]({{urlRoot}}/reference/glossary#spatialos-world) when you [deploy]({{urlRoot}}/reference/glossary#deploying).
```

Maybe? to highlight that we don't have to pass in a snapshot to start a cloud dpl?

File: docs/reference/concepts/snapshots.md [hunk 3]
```diff
@@ -2,32 +2,52 @@
...
+public static void GenerateSnapshot() 
+{
+    // Create a snapshot object
+    var snapshot = new Snapshot();
 
-You can use this as a base for your own project’s snapshot generation by copying the file to the same folder in your own project and editing the `SnapshotGenerator` class to add SpatialOS entities to a snapshot.
+    // Create a template...
+    var template = new EntityTemplate();
+    template.AddComponent(new Position.Snapshot(playerSpawnerLocation), WorkerUtils.UnityGameLogic);
+    template.AddComponent(new Metadata.Snapshot { EntityType = "Cube" }, WorkerUtils.UnityGameLogic);
```
Comment (paulbalaji) [inline docs/reference/concepts/snapshots.md]:
```suggestion
    template.AddComponent(new Metadata.Snapshot("Cube"), WorkerUtils.UnityGameLogic);
```

File: docs/reference/concepts/snapshots.md [hunk 4]
```diff
@@ -2,32 +2,52 @@
...
+}
+```
 
-This saves the generated snapshot to `snapshots/default.snapshot`, which is where SpatialOS expects to find it unless explicitly told to use another path when you start the deployment.
+A common usage pattern is to expose the snapshot generation through a Unity Editor menu item or window. You can see an example of this in the [FPS Starter Project](https://github.com/spatialos/gdk-for-unity-fps-starter-project/blob/<%(Var key="current_version")%>/workers/unity/Assets/Fps/Scripts/Editor/SnapshotGenerator/SnapshotMenu.cs). This allows you to quickly generate a snapshot with a few clicks!
 
 ## How to start a deployment from a snapshot
 
-You can start local or cloud deployments using the `spatial local launch` and `spatial cloud launch` commands respectively - see the documentation on [Deploying your game]({{urlRoot}}/reference/concepts/deployments) for details.
+You can start a local deployment via the Unity Editor menu: **SpatialOS** > **Local launch** (Ctrl+L/Cmd+L). This starts a deployment with the snapshot found at `<your_spatialos_project>/snapshots/default.snapshot`. If you wish to launch via the CLI or with a different snapshot, see the section below.
```
Comment (paulbalaji) [inline docs/reference/concepts/snapshots.md]: is it meant to be `your_spatialos_project` or `spatialos_project_root`?

File: docs/reference/concepts/snapshots.md [hunk 5]
```diff
@@ -2,32 +2,52 @@
...
+
+You can start a cloud deployment using [Deployment Launcher Feature Module]({{urlRoot}}/modules/deployment-launcher/overview). This feature module allows you to configure which snapshot you want to start a deployment with.
```
Comment (paulbalaji) [inline docs/reference/concepts/snapshots.md]:
```suggestion
You can start a cloud deployment using the [Deployment Launcher Feature Module]({{urlRoot}}/modules/deployment-launcher/overview). This feature module allows you to configure which snapshot you want to start your cloud deployment with.
```

maybe?

File: docs/reference/concepts/snapshots.md (hunk 2 above)
Comment (jamiebrynes7) [reply docs/reference/concepts/snapshots.md]: It actually _really_ depends how you start it 😬

File: docs/reference/concepts/snapshots.md (hunk 4 above)
Comment (jamiebrynes7) [reply docs/reference/concepts/snapshots.md]: The latter is probably clearer

Comment (jamiebrynes7) [inline docs/reference/concepts/snapshots.md]:
```suggestion
You can start a local deployment via the Unity Editor menu: **SpatialOS** > **Local launch** (Ctrl+L/Cmd+L). This starts a deployment with the snapshot found at `<spatialos_project_root>/snapshots/default.snapshot`. If you wish to launch via the CLI or with a different snapshot, see the section below.
```

**Pull request merged**